--- tests/CMakeLists.txt.orig	2023-04-10 08:15:50
+++ tests/CMakeLists.txt	2023-04-10 08:16:42
@@ -8,6 +8,10 @@
 # (4) Addition of the test executable via add_test.
 
 ### Core tests, which rely only on boost-test and ndarray.
+set(Boost_NO_BOOST_CMAKE OFF)
+set(Boost_USE_STATIC_LIBS OFF)
+set(BUILD_SHARED_LIBS ON)
+
 find_package(Boost COMPONENTS unit_test_framework REQUIRED)
 
 include_directories( ${PROJECT_SOURCE_DIR}/include)
