--- tests/ndarray-eigen.cc.orig	2022-01-10 13:18:40.000000000 -0800
+++ tests/ndarray-eigen.cc	2022-01-10 21:29:45.000000000 -0800
@@ -94,14 +94,15 @@
     Eigen::Matrix<T, Rows, Cols> m3 = m1 * m2;
     for (int i = 0; i < m; ++i) {
         for (int j = 0; j < n; ++j) {
-            BOOST_CHECK(a[i][j] == m3(i, j));
+	     BOOST_CHECK_CLOSE(a[i][j], m3(i, j), 0.000000001);
         }
     }
     Eigen::Array<T, Rows, Cols> m4(m, n);
     m4.setRandom();
     Eigen::Array<T, Rows, Cols> m5 = m4 * asEigenArray(a);
     Eigen::Array<T, Rows, Cols> m6 = m4 * m3.array();
-    BOOST_CHECK((m5 == m6).all());
+    BOOST_CHECK(m5.size() == m6.size());
+    for(int i=0;i< m5.size();i++) BOOST_CHECK_CLOSE(m5(i), m6(i), 0.000000001);
 }
 
 template <typename T, typename XprKind>
